src/routes/middleware/acl.ts

```diff
@@ -0,0 +1,93 @@
+import { exec, ExecException } from 'child_process';
+import { ACL_FILES_PATH, ACL_TOOL_PATH } from '../../config';
+import { promisify } from 'util';
+import e, { NextFunction, Request, Response, Router } from 'express';
+import {
+  fromString as fromHexString,
+} from 'hex-array';
+const execPromise = promisify(exec);
+
+export async function ensureAclPassed(req: Request, res: Response, next: NextFunction) {
+  console.log("[ACL] Control reached middleware ensureAclPassed req.headers.cookie value", req.headers.cookie);
+
+  if (req.headers.cookie) {
```

Comment:
If client doesn't provide the cookie, it will allow to read all the files.
It makes a good backward compatibility with old Apps,
but I think it will be better to always expect the cookie — otherwise to not allow this at all.